dash/include/dash/allocator/internal/Types.h

```diff
@@ -0,0 +1,71 @@
+#ifndef DASH__ALLOCATOR__INTERNAL__TYPES_H__INCLUDED
+#define DASH__ALLOCATOR__INTERNAL__TYPES_H__INCLUDED
+#include <cstddef>
+#include <ostream>
+
+namespace dash {
+namespace allocator {
+
+
+struct memory_block {
+  // Default Constructor
+  memory_block() noexcept
+    : ptr(nullptr)
+    , length(0)
+  {
+  }
+
+  memory_block(void *ptr, size_t length) noexcept
+    : ptr(ptr)
+    , length(length)
+  {
+  }
+
+  // Move Constructor
+  memory_block(memory_block &&x) noexcept { *this = std::move(x); }
+  // Copy Constructor
+  memory_block(const memory_block &x) noexcept = default;
+
+  // Move Assignment
+  memory_block &operator=(memory_block &&x) noexcept
+  {
+    ptr = x.ptr;
+    length = x.length;
+    x.reset();
+    return *this;
+  }
+
+  // Copy Assignment
+  memory_block &operator=(const memory_block &x) noexcept = default;
+
+  /**
+  * The Memory Block is not freed
+  */
+  ~memory_block() {}
+  /**
+   * Clear the memory block
+   */
+  void reset() noexcept
+  {
+    ptr = nullptr;
+    length = 0;
+  }
+
+  /**
+   * Bool operator to make the Allocator code better readable
+   */
+  explicit operator bool() const noexcept
+  {
+    return length != 0 && ptr != nullptr;
+  }
+
+  void *ptr;
```

Comment:
should be encapsulated in `std::unique_ptr`, as otherwise the ptr is not set to `nullptr` on default move.